c/json.c

```diff
@@ -1760,10 +2154,81 @@ int jsonAsBoolean(Json *json) {
 int jsonAsNumber(Json *json) {
   if (json->type == JSON_TYPE_NUMBER) {
     return json->data.number;
+  } else if (json->type == JSON_TYPE_INT64){
+    return json->data.integerValue;
```

Comment:
This will truncate the value if `integerValue > INT_MAX`.

Reply:
Yes.   If you are using V2 types, and you call jsonAsNumber before checking jsonIsFloat, or jsonIsInt64 and using their respective cases, you are "doing it wrong".   jsonAsNumber is really a legacy / backward-compatibility.   All current code aside from configmgr does NOT use V2 semantics.